src/components/Composer.vue

```diff
@@ -430,14 +470,15 @@ export default {
 				})
 		},
 		reset() {
-			this.selectTo = []
-			this.selectCc = []
-			this.selectBcc = []
-			this.subjectVal = ''
-			this.bodyVal = ''
-			this.attachments = []
-			this.errorText = undefined
-			this.state = STATES.EDITING
+			Object.assign(this.$data, this.$options.data.apply(this))
```

Comment:
is it safe to assign new props to [this.$data](https://vuejs.org/v2/api/#data)? I can't find any docs on `this.$options`. Is this even an official API?

Reply:
> I can't find any docs on this.$options. Is this even an official API?

[It is](https://vuejs.org/v2/api/#vm-options)

> is it safe to assign new props to this.$data? 

Well, I took this approach from the [Vue issue tracker](https://github.com/vuejs/vue/issues/702#issuecomment-308991548), so I wouldn't consider this neither experimental nor "offically recommended". I can revert back to manually resetting the fields if you feel this is more safe.

Reply:
Oh, okay. I would prefer the manual way indeed.

Reply:
Sure! I reverted the change.